encoding/thrift/multiplex.go

```diff
@@ -48,3 +49,46 @@ func (m multiplexedOutboundProtocol) DecodeEnveloped(r io.ReaderAt) (wire.Envelo
 	e.Name = strings.TrimPrefix(e.Name, m.Service+":")
 	return e, err
 }
+
+type multiplexedOutboundNoWireProtocol struct {
+	stream.Protocol
+
+	Service string
+}
+
+func (m multiplexedOutboundNoWireProtocol) Writer(w io.Writer) stream.Writer {
+	return multiplexedWriter{
+		Writer:  m.Protocol.Writer(w),
+		Service: m.Service,
+	}
+}
+
+func (m multiplexedOutboundNoWireProtocol) Reader(r io.Reader) stream.Reader {
+	return multiplexedReader{
+		Reader:  m.Protocol.Reader(r),
+		Service: m.Service,
+	}
+}
+
+type multiplexedWriter struct {
+	stream.Writer
+
+	Service string
+}
+
+func (w multiplexedWriter) WriteEnvelopeBegin(eh stream.EnvelopeHeader) error {
+	eh.Name = w.Service + ":" + eh.Name
+	return w.Writer.WriteEnvelopeBegin(eh)
+}
+
+type multiplexedReader struct {
+	stream.Reader
+
+	Service string
+}
+
+func (r multiplexedReader) ReadEnvelopeBegin() (stream.EnvelopeHeader, error) {
+	eh, err := r.Reader.ReadEnvelopeBegin()
+	eh.Name = strings.TrimPrefix(eh.Name, r.Service+":")
+	return eh, err
+}
```

Comment:
Why do multiplexedReader and multiplexedWriter does not have `WriteEnvelopeEnd` and `ReadEnvelopeEnd` ?